example/main.go

```diff
@@ -106,19 +112,74 @@ func Main() int {
 		case errors.Is(err, io.EOF):
 			log.Infof("EOF received, exiting.")
 			return 0
+		case errors.Is(err, terminal.ErrInterrupted):
+			interrupts++
+			if interrupts >= 3 {
+				log.Infof("Triple interrupt, exiting.")
+				return 0
+			}
+			log.Infof("Interrupted, resetting, use exit or Ctrl-d. to exit.")
+			ctx, cancel = t.ResetInterrupts(context.Background()) //nolint:fatcontext // this is only upon interrupt.
 		default:
 			return log.FErrf("Error reading line: %v", err)
 		}
-		log.Infof("Read line got: %q", cmd)
 		// Save previous command validity to know whether this one should replace it in history or not.
 		previousCommandWasValid = isValidCommand
 		isValidCommand = false // not valid unless proven otherwise (reaches the end validations etc)
+		if cmd == "" {
+			// we do get empty reads on interrupt
+			log.LogVf("Empty command read")
+			continue
+		}
+		log.Infof("Read line got: %q", cmd)
 		switch {
 		case cmd == exitCmd:
+			log.Infof("Exit command received, exiting.")
 			return 0
 		case cmd == helpCmd:
 			fmt.Fprintf(t.Out, "Available commands: %v\n", commands)
 			isValidCommand = true
+		case strings.HasPrefix(cmd, sleepCmd):
+			parts := strings.SplitN(cmd, " ", 2)
+			if len(parts) < 2 {
+				fmt.Fprintf(t.Out, "Usage: %s <duration>\n", sleepCmd)
+				continue
+			}
+			dur, err := time.ParseDuration(parts[1])
+			if err != nil {
+				fmt.Fprintf(t.Out, "Invalid duration %q: %v\n", parts[1], err)
+				continue
+			}
+			isValidCommand = true
+			log.Infof("Sleeping for %v (^C to interrupt)", dur)
+			err = terminal.SleepWithContext(t.Context, dur)
+			if err != nil {
+				log.Infof("Sleep interrupted: %v", err)
+				interrupts = 0
+			}
+		case strings.HasPrefix(cmd, cancelCmd):
+			parts := strings.SplitN(cmd, " ", 2)
+			if len(parts) < 2 {
+				fmt.Fprintf(t.Out, "Usage: %s <duration>\n", cancelCmd)
+				continue
+			}
```

Comment:
You know it starts with sleepCmd

Why not using strings.TrimPrefix ?

Reply:
because of the history, also why mutate/remove the prefix even if I didn't need to keep it as entered?